compile/compile.go

```diff
@@ -697,3 +733,67 @@ func (ss orderedStringSet) Append(s ...string) orderedStringSet {
 	}
 	return ss
 }
+
+func stringsToRefs(x []string) []ast.Ref {
+	result := make([]ast.Ref, len(x))
+	for i := range result {
+		result[i] = storage.MustParsePath("/" + x[i]).Ref(ast.DefaultRootDocument)
+	}
+	return result
+}
+
+type refSet struct {
+	s ast.Set
+}
+
+func newRefSet(x ...ast.Ref) *refSet {
+	result := &refSet{s: ast.NewSet()}
+	for i := range x {
+		result.AddPrefix(x[i])
+	}
+	return result
+}
+
+// ContainsPrefix returns true if r is prefixed by any of the existing refs in the set.
+func (rs *refSet) ContainsPrefix(r ast.Ref) bool {
+	return rs.s.Until(func(x *ast.Term) bool {
+		exist := x.Value.(ast.Ref)
```

Comment:
This cast can't fail, I suppose...?

Reply:
Correct...however, I just realized this could be implemented simply on top of a slice. No need to use ast.Set here. That avoids the need to type assert. Thanks!